controllers/buildpipeline/buildpipeline_controller.go

```diff
@@ -137,6 +139,7 @@ func setupControllerWithManager(manager ctrl.Manager, controller *Reconciler) er
 	return ctrl.NewControllerManagedBy(manager).
 		For(&tektonv1beta1.PipelineRun{}).
 		WithEventFilter(predicate.Or(
-			tekton.BuildPipelineRunSignedAndSucceededPredicate())).
+			tekton.BuildPipelineRunSignedAndSucceededPredicate(),
+			tekton.BuildPipelineRunCreatedPredicate())).
```

Comment:
I think that we'll also need to catch failed build pipelineRuns in order to remove the finalizer, and the current predicates do not catch those events.

Reply:
You're right, my push failed and I wasn't paying attention.  I've just pushed a change with a new predicate called `BuildPipelineRunFailedPredicate()`.